cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -60,6 +60,10 @@ func NewCloudControllerManagerOptions() *CloudControllerManagerOptions {
 
 // AddFlags adds flags for a specific ExternalCMServer to the specified FlagSet
 func (o *CloudControllerManagerOptions) AddFlags(fs *pflag.FlagSet) {
+	if o == nil {
```

Comment:
How could this be nil?

`NewCloudControllerManagerOptions` won't return nil `CloudControllerManagerOptions`.

Reply:
You are right. Mostly are `standard of programming`, using pointer.

Reply:
If `o` is nil, panic will occur first before calling methods, like `AddFlags`. That means `if o == nil` in L63 will never been run, since `nil.AddFlags` panic first.

I don't think current change make any sense.

Reply:
@dixudx You are right. In this repo, all codes are controlled. But the struct and also its methods are public, this protection can protect outside usage.